tests/tabs.ts

```diff
@@ -0,0 +1,387 @@
+        expect(tabs._internal_fashion).to.equal('accordion');
+
+        await setViewport({width:1000, height:640});
+        create_slot_el(tabs);
+        window.dispatchEvent(new Event('resize'));
+        expect(tabs._internal_fashion).to.equal('row');
+        expect(validate_el(tabs, 2)).to.equal(true);
+
+        await setViewport({width:1640, height:680});
+        window.dispatchEvent(new Event('resize'));
+        validate_RC(tabs, 5);
+        expect(tabs._internal_fashion).to.equal('column');
+        expect(tabs._column_size.title).to.equal('30%');
+        expect(tabs._column_size.body).to.equal('70%');
+
+        let col = "title:50%;body:50%";
+        tabs.setAttribute("closize", col);
+        expect(tabs._column_size.title).to.equal('30%');
+        expect(tabs._column_size.body).to.equal('70%');
+        tabs.attributeChangedCallback('colsize', '', col);
+        expect(tabs._column_size.title).to.equal('50%');
+        expect(tabs._column_size.body).to.equal('50%');
+
+        await setViewport({width:600, height:640});
+        const el2 = await fixture (`
+            <div id='parent'>
+                <aalam-tabs animation="r2l" 
+                            fashion="xs:accordion;m:row;xl:column">
+                    <div slot="tab-title" id="t1">title1</div>
+                    <div slot="tab-body" id='b1'>body1</div>
+                    <div slot='tab-title' id='t2'>title2</div>
+                    <div slot='tab-body' id='b2'>body2</div>
+                    <div slot="tab-title" id="t3">title3</div>
+                    <div slot="tab-body" id='b3'>body3</div>
+                    <div slot='tab-title' id='t4'>title4</div>
+                    <div slot='tab-body' id='b4'>body4</div>
+                </aalam-tabs>
+            </div>
+        `);
+        let tab2 = el2.querySelector("aalam-tabs");
+        expect(tab2._internal_fashion).to.equal('accordion');
+        validate_acc(tab2, 4);
+
+        await setViewport({width:1240, height:640});
+        window.dispatchEvent(new Event('resize'));
+        validate_RC(tab2, 4);
+        expect(tab2._internal_fashion).to.equal('row');
+
+        await setViewport({width:1680, height:640});
+        create_slot_el(tab2);
+        window.dispatchEvent(new Event('resize'));
+        validate_RC(tab2, 5);
+        expect(tab2._internal_fashion).to.equal('column');
+        expect(tab2._column_size.title).to.equal('30%');
+        expect(tab2._column_size.body).to.equal('70%');
+
+        await setViewport({width:800, height:640});
+        create_slot_el(tab2);
+        window.dispatchEvent(new Event('resize'));
+        expect(tab2._internal_fashion).to.equal('accordion');
+        validate_acc(tab2, 6);
+
+        await setViewport({width:1500, height:640});
+        window.dispatchEvent(new Event('resize'));
+        validate_RC(tab2, 6);
+        expect(tab2._internal_fashion).to.equal('row');
+
+        await setViewport({width:1100, height:640});
```

Comment:
In all these cases and beneath it, it will be good to validate which element is actually shown on each change.